async_simple/coro/Collect.h

```diff
@@ -507,6 +507,45 @@ inline auto CollectAnyVariadicImpl(LazyType<Ts>&&... inputs) {
     return AT(std::move(inputs)...);
 }
 
+// collectAny with std::pair<Lazy, CallbackFunction>
+template <typename Tuple>
+class Helper {
+public:
+    Helper(Tuple tuple) : _callback_tuple(std::move(tuple)) {}
+
+    async_simple::coro::Lazy<size_t> operator()(auto&&... lazy) {
+        auto result = co_await collectAny(std::move(lazy)...);
+
+        tupleSwitch(result.index(), _callback_tuple, result);
+        co_return result.index();
```

Comment:
Let's try to avoid adding layers in implementing fundamental functions by new co_await. It may results new allocations/deallocations, which may be a performance concern.